src/Microsoft.ComponentDetection.Detectors/nuget/NuGetProjectModelProjectCentricComponentDetector.cs

```diff
@@ -264,19 +103,27 @@ private void NavigateAndRegister(
         ISingleFileComponentRecorder singleFileComponentRecorder,
         LockFileTargetLibrary library,
         string parentComponentId,
-        HashSet<string> dotnetRuntimePackageNames,
+        FrameworkPackages frameworkPackages,
         HashSet<string> visited = null)
     {
-        if (this.IsAFrameworkComponent(dotnetRuntimePackageNames, library.Name, library.Dependencies)
-            || library.Type == ProjectDependencyType)
+        if (library.Type == ProjectDependencyType)
         {
             return;
         }
 
+        var isFrameworkComponent = frameworkPackages.IsAFrameworkComponent(library.Name, library.Version);
+        var isDevelopmentDependency = IsADevelopmentDependency(library);
+
         visited ??= [];
 
         var libraryComponent = new DetectedComponent(new NuGetComponent(library.Name, library.Version.ToNormalizedString()));
-        singleFileComponentRecorder.RegisterUsage(libraryComponent, explicitlyReferencedComponentIds.Contains(libraryComponent.Component.Id), parentComponentId);
+        singleFileComponentRecorder.RegisterUsage(libraryComponent, explicitlyReferencedComponentIds.Contains(libraryComponent.Component.Id), parentComponentId, isDevelopmentDependency: isFrameworkComponent || isDevelopmentDependency);
+
+        // get the actual component in case it already exists
+        libraryComponent = singleFileComponentRecorder.GetComponent(libraryComponent.Component.Id);
+
+        // Add framework information to the actual component
+        ((NuGetComponent)libraryComponent.Component).TargetFrameworks.Add(target.TargetFramework.GetShortFolderName());
```

Comment:
ditto about being certain that TargetFramework is populated everywhere

Reply:
I think we can trust it here - this isn't a user file - but I will double check if I can somehow break it.

Reply:
For reference here's the place it's set https://github.com/NuGet/NuGet.Client/blob/8b88f9e28677f39bd871f4edded2ea4f68b01ac7/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileFormat.cs#L404

I think we can count on it always being set, nonetheless I added a fallback.